controllers/iamauthpolicy_controller.go

```diff
@@ -139,21 +139,32 @@ func (c *IAMAuthPolicyController) findSnId(ctx context.Context, k8sPolicy *anv1a
 	return *snInfo.SvcNetwork.Id, nil
 }
 
-func (c *IAMAuthPolicyController) upsertGatewayPolicy(ctx context.Context, k8sPolicy *anv1alpha1.IAMAuthPolicy) (string, error) {
+func (c *IAMAuthPolicyController) upsertGatewayPolicy(ctx context.Context, k8sPolicy *anv1alpha1.IAMAuthPolicy) error {
+	err := c.handleConflicts(ctx, k8sPolicy)
+	if err != nil {
+		return err
+	}
 	snId, err := c.findSnId(ctx, k8sPolicy)
```

Comment:
Just check LatticeSN/Serve resource to determine whether update Policy status to `gwv1alpha2.PolicyReasonTargetNotFound` looks a smart way, I think it works well?

Reply:
didnt get the question

Reply:
The [accessLogPolicy](https://github.com/aws/aws-application-networking-k8s/blob/a2d5b973b3516f11784d1a764830e8de2eadf4a7/controllers/accesslogpolicy_controller.go#L214), check ref k8s resource exist or not to update alp with PolicyReasonTargetNotFound, (and check lattice resource later again in buildAndDeployModel and [do nothing if latticeResource not found?](https://github.com/aws/aws-application-networking-k8s/blob/de8994c8ffbab778f64ab9fe26f3e82f0bdfa6da/pkg/deploy/lattice/access_log_subscription_manager.go#L46), I didn't quite get this part ALP logic)

Anyway, I think this PR this part is good

Reply:
good point, probably PolicyReasonTargetNotFound should only check for k8s reference not lattice